### PR TITLE
docs: add links to figma assets part 3

### PR DESCRIPTION
**Pull request by marissahuysentruyt**
## Description

This PR is the third in a series that adds S2 design links to components. Now, when a user is on a story page (not a docs page), under the "Design" tab, a Figma asset should render. 

Screenshot 2024-10-04 at 10 58 33 AM

Using the [S2 / Desktop Figma file](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=5634-5140&node-type=frame&t=YT3sYHqnhqpnjdv9-0), the top-level component pages are linked in an attempt to give a user (specifically a developer) context to the component they are viewing. (instructions on how to get the Figma frame links are here: [slack canvas](https://adobe.enterprise.slack.com/docs/T024FSURM/F07PTPUPMRV)

Components affected:
- standard/default dialog
- takeover dialog
- divider
- drop zone
- field label
- help text
- illustrated message
- inline alert
- link
- menu (menu group in Figma = default menu; menu item in Figma = menu item story/variant)
- meter
- picker

_Note: not all Spectrum CSS components have corresponding Figma designs._

### Jira/Specs

[CSS-974](https://jira.corp.adobe.com/browse/CSS-974)

#### Design pages
- [standard dialog](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=839-1128)
- [takeover dialog](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=61935-5399)
- [divider](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=13642-334)
- [drop zone](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=13049-185)
- [field label](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=16090-95)
- [help text](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=13653-196)
- [illustrated message](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=20032-601)
- [inline alert](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=2659-4482)
- [link](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=18850-110)
- [menu](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=37252-553)
- [menu item](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=37252-2242&node-type=frame&t=Kcz7zeePp3PeRusJ-11)
- [meter](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=33840-90)
- [picker](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=739-1453)

### Pending Questions

- Are there any objections to how the dialog stories and menu stories are linked to Figma? Menu sort of breaks the convention of adding a link to the top-level component page, with `Menu item` linked to a certain frame in the design.
- Is there any icon Figma file we could link to? 

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [ ] Pull down the branch to run locally 
- [ ] Visit [the dialog default story page](http://localhost:8080/?path=/story/components-dialog--default)
- [ ] Open the `Design` tab (in the add-ons control panel)
- [ ] Verify a corresponding S2 Figma asset for the component now renders. It should say which page and which frame was rendered in the bottom left corner.

Screenshot 2024-10-04 at 11 35 47 AM

- [ ] Repeat the steps above for each of the affected components listed
    - [ ] default dialog (called standard dialog in Figma
    - [ ] fullscreen dialog (a variant of takeover dialog in Figma)
    - [ ] fullscreen takeover dialog (a variant of takeover dialog in Figma)
    - [ ] divider
    - [ ] drop zone
    - [ ] field label
    - [ ] help text
    - [ ] illustrated message
    - [ ] inline alert
    - [ ] link
    - [ ] default menu (called menu group in Figma)
    - [ ] menu item (called menu item in Figma)
    - [ ] meter
    - [ ] picker

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/dialog/stories/dialog.stories.js [hunk 1]
```diff
@@ -119,6 +119,10 @@ export default {
 				height: "500px",
 			},
 		},
+		design: {
```
Comment (castastrophe) [inline components/dialog/stories/dialog.stories.js:122]: Oh good call!  This is great!

File: components/fieldlabel/stories/form.template.js [hunk 2]
```diff
@@ -43,21 +43,11 @@ export const Template = ({
                         alignment: labelPosition === "side" ? fieldLabelAlignment : undefined,
                     }, context))}
                     <div class=${classMap({
-                        [`${rootClass}-item`]: true,
+                        [`${rootClass}-itemField`]: true,
                     })}>
-                        ${when(label, () => FieldLabel({
```
Comment (castastrophe) [inline components/fieldlabel/stories/form.template.js:48]: This is a great update but feels out of scope for this work. Would you mind opening a PR for this off main?

**Review by castastrophe — CHANGES_REQUESTED** [review]

File: components/fieldlabel/stories/form.template.js (hunk 2 above)
Comment (marissahuysentruyt) [reply components/fieldlabel/stories/form.template.js]: Good call, this has been dropped. 

I only see this duplicate code on the s2-foundations branches. Right now, I have a draft PR, but it's against `s2-foundations-design-links`. I'll rebase and get that ready for `s2-foundations-redux`.

**Review by castastrophe — APPROVED** [review]
